src/scheduler/fifo.c

```diff
@@ -695,6 +699,11 @@ scheduling_cycle(int sd, char *jobid)
 
 	update_cycle_status(&cstat, 0);
 
+#ifdef PBS_UNDOLR_ENABLED
+	if (sigusr1_flag)
+		undolr();
```

Comment:
in case it is necessary to use a global and check it every time, we should do that in main() and not here

Reply:
I could not find the re-entrant place in main () for this call.

Reply:
i am not sure why you need re-entrant place.
how about the infinite for loop in main()

Reply:
Done, now recording will start before starting scheduling cycle.